Public/Src/Utilities/Collections/BitSet.cs

```diff
@@ -100,7 +100,10 @@ private bool TryCompareExchangeEntry(int index, ulong value, ulong comparand)
         /// </summary>
         public void Add(int index)
         {
-            Contract.Requires(index >= 0 && index < Length);
+            if (index < 0 || index >= Length)
+            {
+                Contract.Requires(false, $"index={index} must be within [0, {Length}) range.");

```

Comment:
Did you mean a ] instead of ) ?
> $"index={index} must be within [0, {Length}) range.");

$"index={index} must be within [0, {Length}**]** range.");

Reply:
No, the supported range is [0, Length), i.e., Length is not included.